server/app/controllers/applicant/ApplicantProgramReviewController.java

```diff
@@ -106,12 +117,41 @@ public CompletionStage<Result> submit(Request request, long applicantId, long pr
                 if (cause instanceof SecurityException) {
                   return unauthorized();
                 }
+                if (cause instanceof ApplicationOutOfDateException) {
+                  Call reviewPage =
+                      routes.ApplicantProgramReviewController.review(applicantId, programId);
+                  return redirect(reviewPage).flashing("error", cause.getMessage());
+                }
                 throw new RuntimeException(cause);
               }
               throw new RuntimeException(ex);
             });
   }
 
+  /**
+   * Validates that the application is complete and correct to submit.
+   *
+   * <p>An application may be submitted but incomplete for a variety of reason:
+   *
+   * <ul>
+   *   <li>The application view with submit button contains stale data that has changed visibility
+   *       or eligibility conditions that result in the application being incomplete or ineligible.
+   * </ul>
+   *
+   * @return a {@link ApplicationOutOfDateException} wrapped in a failed future with a user visible
+   *     message for the issue.
+   */
+  private CompletableFuture<Void> validateApplication(
```

Comment:
suggestion (blocking): given the validation is specific to submitting, recommend renaming to `validateApplicationForSubmission`

Reply:
done